mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1307,7 +1318,7 @@ pub async fn gen_and_sign_taker_payment_spend_preimage<T: UtxoCommonOps>(
 
     let sig_hash_type = match args.dex_fee {
         DexFee::Standard(_) => SIGHASH_SINGLE,
-        DexFee::WithBurn { .. } => SIGHASH_ALL,
+        DexFee::WithBurn { .. } | DexFee::Zero => SIGHASH_ALL,
```

Comment:
Could you clarify why we should use SIGHASH_ALL for zero fee?

Reply:
ok I was just curious will it be the difference for zero fee, as for `Standard` dex fee we use `SIGHASH_SINGLE`

Reply:
Yes, we should use SIGHASH_ALL wherever possible.
SIGHASH_SINGLE could be used only in a special case (DexFee::Standard) when maker can add an output to the spending taker payment transaction (with only a single dexfee output signed by the  taker with the SIGHASH_SINGLE flag), without the other party signature.